mlflow/langchain/utils.py

```diff
@@ -422,6 +436,56 @@ def _get_path_by_key(root_path, key, conf):
     return os.path.join(root_path, key_path) if key_path else None
 
 
+def _patch_loader(loader_func: Callable) -> Callable:
+    """
+    Patch LangChain loader function like load_chain() to handle the breaking change introduced in
+    LangChain 0.1.12.
+
+    Since langchain-community 0.0.27, loading a module that relies on the pickle deserialization
+    requires the `allow_dangerous_deserialization` flag to be set to True, for security reasons.
+    However, this flag could not be specified via the LangChain's loading API like load_chain(),
+    load_llm(), until LangChain 0.1.14. As a result, such module cannot be loaded with MLflow
+    with earlier version of LangChain and we have to tell the user to upgrade LangChain to 0.0.14
+    or above.
+
+    Args:
+        loader_func: The LangChain loader function to be patched e.g. load_chain().
+
+    Returns:
+        The patched loader function.
+    """
+    if not IS_PICKLE_SERIALIZATION_RESTRICTED:
+        return loader_func
+
+    import langchain
+
+    if Version(langchain.__version__) >= Version("0.1.14"):
+        # For LangChain 0.1.14 and above, we can pass `allow_dangerous_deserialization` flag
+        # via the loader APIs. Since the model is serialized by the user (or someone who has
+        # access to the tracking server), it is safe to set this flag to True.
+        def patched_loader(*args, **kwargs):
+            return loader_func(*args, **kwargs, allow_dangerous_deserialization=True)
```

Comment:
Any chance that `kwargs` already has `allow_dangerous_deserialization`?

Reply:
It looks like there is no chance for `kwargs` to include `allow_dangerous_deserialization`.

Reply:
Yup!